### DISCUSSION TITLE
Toggle Switch via parent component

### DISCUSSION THREAD

**Discussion by devklick** [mantinedev/mantine#5958] [Q&A] [answered]
I'm trying to create a `Menu.Item` that acts as a color scheme toggle, and I want to use the `leftSection` as a `Switch` that displays `onLabel` & `offLabel` sun/moon icons, e.g:
```jsx
const { toggleColorScheme } = useMantineColorScheme();
const computedColorScheme = useComputedColorScheme();
const themeSwitch = (
  <Switch
    checked={computedColorScheme === "light"}
    onChange={toggleColorScheme}
    onLabel={<IconSun />}
    offLabel={<IconMoon />}
  />
);
return (
  <Menu shadow="md" width={200} trigger="hover">
    <Menu.Target>
      <Button>Settings</Button>
    </Menu.Target>
    <Menu.Dropdown>
      <Menu.Item onClick={toggleColorScheme} leftSection={themeSwitch}>
        Theme
      </Menu.Item>
    </Menu.Dropdown>
  </Menu>
```

![image](https://github.com/mantinedev/mantine/assets/58235000/f480898b-b515-4554-855c-acd18fe776da)

Clicking on the `Menu.Item` works and changes the color scheme, however clicking on the `Switch` does not - nothing happens. It seems like the click event is being fired on both the Switch and the Menu.Item, so presumably they're cancelling each other out. 

Is it possible to use a switch like this, as a section of the menu item, and have the switches checked value controlled via clicking *anywhere* within the menu item?

References: 
- [Menu - Show On Hover](https://mantine.dev/core/menu/#show-menu-on-hover)
- [Switch - Icon Labels](https://mantine.dev/core/switch/#icon-labels)

**Comment by rtivital** [comment, marked as answer]
1. It is an accessibility violation to render a checkbox inside a menu item because the menu item is a button
2. If want do disabled click events on the Switch component, set `style={{ pointerEvents: 'none' }}`

**Comment by devklick** [comment]
Thanks @rtivital, for the advice and the solution. 

I intend to use the switch as a visual element only, not clickable, so this solution is perfect for me.
